lib/spack/spack/environment/environment.py

```diff
@@ -467,6 +468,9 @@ def from_dict(base_path, d):
 
     @property
     def _current_root(self):
+        if spack.util.atomic_update.use_renameat2():
```

Comment:
I think whether we *use* it should be made into a module-scope var or function *here*, and that should ask `atomic_update` whether it's actually available.